### PR TITLE
Check result of ECR scan and notify on slack if results returned

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
## Purpose
In order to prevent known vulnerabilities making it onto production, we will (as a minimum) scan container images using ECR scan-on-push. The pipeline will wait for results and check the findings, failing on a non-successful result.

Fixes UML-540

## Approach

Use boto3 to request results, and use requests and webhooks to poost messages to slack

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: pipeline_scripts/check_ecr_scan_results/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Check AWS ECR scan results
+
+This script returns ECR scan results for known vulnerabilities.
+
+Information about ECR scan on push is available at <https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-scanning.html>
+
+The script takes arguments for image tag to return results for, the Slack webhook to use for posting results and whether to post to slack.
+
+If ommitted, the image tag will default to `latest`, and the webhoook will default to the system environment variable `$SLACK_WEBHOOK`.
```
Comment (williamfalconeruk) [inline pipeline_scripts/check_ecr_scan_results/README.md]: webhoook - typo

Comment (williamfalconeruk) [reply pipeline_scripts/check_ecr_scan_results/README.md]: /pedantic :)

Comment (andrewpearce-digital) [reply pipeline_scripts/check_ecr_scan_results/README.md]: why is my spell checker not finding these!??

File: pipeline_scripts/check_ecr_scan_results/aws_ecr_scan_results.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+            'ecr',
+            region_name='eu-west-1',
+            aws_access_key_id=self.aws_iam_session['Credentials']['AccessKeyId'],
+            aws_secret_access_key=self.aws_iam_session['Credentials']['SecretAccessKey'],
+            aws_session_token=self.aws_iam_session['Credentials']['SessionToken'])
+
+    def read_parameters_from_file(self, config_file):
+        with open(config_file) as json_file:
+            parameters = json.load(json_file)
+            self.aws_account_id = 311462405659
```
[the remark this answers is not in the document]
Comment (andrewpearce-digital) [reply pipeline_scripts/check_ecr_scan_results/aws_ecr_scan_results.py]: This is the management account id where all our ECR images live for all UAL accounts

File: pipeline_scripts/check_ecr_scan_results/README.md (hunk 1 above)
Comment (andrewpearce-digital) [reply pipeline_scripts/check_ecr_scan_results/README.md]: fixed those

**Pull request merged**